Try to generate usage completions and it fails

```cmd
❯ mise completion clink
Generating completions ...
[ERROR] failed to find bash shell! Try again with inputting the shell flag
[ERROR] failed to generate completions (path:C:\Users\KUCOLE\workspace\github\mise-clink\mise.usage.lua); exit code: 1
```

```cmd
❯ tree /F bin
Folder PATH listing for volume OSDisk
Volume serial number is C45C-9E1B
C:\USERS\KUCOLE\WORKSPACE\GITHUB\MISE-CLINK\BIN
    cuc-v0.2.0.exe
    cuc.exe
``` 

```cmd
❯ bin\cuc.exe 
A CLI tool to generate clink argmatcher completions from the usage spec.

Usage: cuc.exe [COMMAND]

Commands:
  generate  Generate clink argmatcher completions from the usage spec
  complete  Get completions by running the specified command as args
  usage     Generate usage.kdl for this CLI itself
  help      Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
```

```cmd
❯ clink --version
1.7.6.d9907d

mise-clink on  main via 🌙 
❯ mise --version
              _                                        __
   ____ ___  (_)_______        ___  ____        ____  / /___ _________
  / __ `__ \/ / ___/ _ \______/ _ \/ __ \______/ __ \/ / __ `/ ___/ _ \
 / / / / / / (__  )  __/_____/  __/ / / /_____/ /_/ / / /_/ / /__/  __/
/_/ /_/ /_/_/____/\___/      \___/_/ /_/     / .___/_/\__,_/\___/\___/
                                            /_/                 by @jdx
2025.9.19 windows-x64 (2025-09-25)
```

```
❯ mise completion clink
Generating completions ...
[ERROR] failed to find bash shell! Try again with inputting the shell flag
[ERROR] failed to generate completions (path:C:\Users\KUCOLE\workspace\github\mise-clink\mise.usage.lua); exit code: 1
```

> it works If I add --shell cmd:
```
❯ bin\cuc.exe generate --complete --shell cmd mise.usage.latest.kdl >mise.usage.lua
```

Refer to 3rd point: https://github.com/IMXEren/cuc?tab=readme-ov-file#installation

--- a/mise.lua
+++ b/mise.lua
@@ -469,10 +469,10 @@ end
 
 --------------------------------------------------------------------------------
 -- Generate mise usage completions
--- To use it, you'd have to run `mise completion clink`
+-- To use it, you'd have to run `mise completion clink -- [OPTIONS] [SUBCOMMANDS] [ARGS]`
 -- The completions are stored in the same directory as mise.cmd
 --------------------------------------------------------------------------------
-local function completion()
+local function completion(args)
     -- Generate mise usage completions
     local Cuc = require("cuc")
     local mise_cmd_bin_dir = path.join(mise_cmd_dir, "bin")
@@ -492,7 +492,7 @@ local function completion()
     local mise_completions_lua = path.join(mise_cmd_dir, "mise.usage.lua")
     if cuc:check_cuc() then
         print("Generating completions ...")
-        local ok, code, completions = cuc:generate_completions(mise_path)
+        local ok, code, completions = cuc:generate_completions(mise_path, args)
         if ok then
             local file = io.open(mise_completions_lua, "w+")
             assert(file, "[ERROR] failed to open file: " .. mise_completions_lua)
@@ -668,7 +668,19 @@ function mise(args)
             os.exit(code)
         end
 
-        local code = completion()
+        local collect_index = #nargs + 1
+        for i, arg in ipairs(nargs) do
+            if arg == "--" then
+                collect_index = i + 1
+                break
+            end
+        end
+        local gargs = {}
+        if #nargs >= collect_index then
+            table.extend(gargs, { table.unpack(nargs, collect_index) })
+        end
+
+        local code = completion(gargs)
         os.exit(code)
     end
 
@@ -730,7 +742,7 @@ if not standalone then
     -- Hook environment variables if mise is activated
     local function _mise_hook()
         if not os.getenv(MISE_ACTIVATED_KEY) then return end
-        local co = coroutine.create(function ()
+        local co = coroutine.create(function()
             hook_env(os.getenv(MISE_HOOK_ENV_ARGS_KEY), nil, true)
         end)
         clink.runcoroutineuntilcomplete(co)

--- a/modules/cuc.lua
+++ b/modules/cuc.lua
@@ -34,8 +34,9 @@ function Cuc:download_cuc()
     return ok, code
 end
 
-function Cuc:generate_completions(mise_path)
-    local cmd_line = string.format([[""%s" usage | "%s" generate --complete"]], mise_path, self.path)
+function Cuc:generate_completions(mise_path, args)
+    args = args or {}
+    local cmd_line = string.format([[""%s" usage | "%s" generate --complete %s"]], mise_path, self.path, '"' .. table.concat(args, '" "') .. '"')
     local p = io.popen(cmd_line)
     assert(p, "[ERROR] failed to generate completions: " .. (cmd_line or "(nil)"))
     local output = p:read("*a")

